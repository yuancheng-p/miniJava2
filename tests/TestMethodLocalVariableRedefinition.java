public class A {
    int a5 = 1;
    int[] a = { 3,4,5 };
    boolean a2 = false;

    public void fooA(){
    
    int e3 = 8;
    int g1 = 6;
    int y1;
}
// fooB will not pass
    public void fooB(){
    
    int e13 = 8;
    int q1 = 6;
    boolean q1 = true;
}

}

public class B {
    int a5 = 1;
    int[] a = { 3,4,5 };
    boolean a2 = false;

    public void fooA(){
    
    int e8 = 8;
    int g4 = 6;
    int y0;
}
// fooB will not pass
    public void fooB(){
    
    int e1 = 8;
    int e1;
}

}



