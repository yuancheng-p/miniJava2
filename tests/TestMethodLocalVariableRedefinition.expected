successfull parsing
start typing
check add and methods:A:fooA
check add and methods:A:fooA1
check add and methods:A:fooB
check add and methods:A:fooC
check add and methods:A:fooD
check add and methods:B:fooA
check add and methods:B:fooB
--------- env --------
A : class_env(parent=: Object)
method : fooB
method : fooA1 paramlist: boolean  paramlist: int  paramlist: int 
method : fooA
method : fooD paramlist: boolean  paramlist: int  paramlist: int 
method : fooC paramlist: boolean  paramlist: int  paramlist: int 
 attr:  boolean a2 = false;
 attr:  int[] a = {3,4,5};
 attr:  int a5 = 1;
B : class_env(parent=: Object)
method : fooB
method : fooA
 attr:  boolean a2 = false;
 attr:  int[] a = {3,4,5};
 attr:  int a5 = 1;
Boolean : class_env(parent=: Object)
Integer : class_env(parent=: Object)
Object : class_env(parent=: )
String : class_env(parent=: Object)
